runtime-modules/storage/src/lib.rs

```diff
@@ -4459,97 +4444,110 @@ impl<T: Trait> Module<T> {
         Ok(())
     }
 
-    fn construct_objects_to_upload(
-        op: &BagOperation<T>,
-    ) -> Result<Vec<DataObject<BalanceOf<T>>>, DispatchError> {
+    fn construct_objects_to_upload(op: &BagOperation<T>) -> DataObjAndDeletionPrizeAndObjSize<T> {
         match &op.params {
-            BagOperationParams::<T>::Delete => Ok(Default::default()),
-            BagOperationParams::<T>::Create(_, list, _, _) => {
-                Self::construct_objects_from_list(list)
+            BagOperationParams::<T>::Delete => Ok((Default::default(), Zero::zero(), Zero::zero())),
+            BagOperationParams::<T>::Create(bag_deletion_prize, list, _, _) => {
+                Self::construct_objects_from_list(list, *bag_deletion_prize)
+            }
+            BagOperationParams::<T>::Update(list, _) => {
+                Self::construct_objects_from_list(list, Zero::zero())
             }
-            BagOperationParams::<T>::Update(list, _) => Self::construct_objects_from_list(list),
         }
     }
 
     fn construct_objects_from_list(
```

Comment:
This function will probably change in a future refactoring since it has became quite complex.
For now please just add a doc line like describing the main units of work done